packages/protocol/contracts/bridge/erc721/libs/LibERC721.sol

```diff
@@ -0,0 +1,208 @@
+// SPDX-License-Identifier: MIT
+//  _____     _ _         _         _
+// |_   _|_ _(_) |_____  | |   __ _| |__ ___
+//   | |/ _` | | / / _ \ | |__/ _` | '_ (_-<
+//   |_|\__,_|_|_\_\___/ |____\__,_|_.__/__/
+
+pragma solidity ^0.8.20;
+
+import {AddressResolver} from "../../../common/AddressResolver.sol";
+import {
+    Create2Upgradeable
+} from "@openzeppelin/contracts-upgradeable/utils/Create2Upgradeable.sol";
+import {
+    ERC721Upgradeable
+} from "@openzeppelin/contracts-upgradeable/token/ERC721/ERC721Upgradeable.sol";
+import {NFTVault} from "../../NFTVault.sol";
+import {BridgedERC721} from "../BridgedERC721.sol";
+import {IBridge} from "../../IBridge.sol";
+import {Strings} from "@openzeppelin/contracts/utils/Strings.sol";
+
+library LibERC721 {
+    /*********************
+     * Events            *
+     *********************/
+
+    event BridgedERC721Deployed(
+        uint256 indexed srcChainId,
+        address indexed canonicalToken,
+        address indexed bridgedToken,
+        string canonicalTokenSymbol,
+        string canonicalTokenName
+    );
+
+    event ERC721Sent(
+        bytes32 indexed msgHash,
+        address indexed from,
+        address indexed to,
+        uint256 destChainId,
+        address token,
+        uint256 tokenId
+    );
+
+    event ERC721Released(
+        bytes32 indexed msgHash,
+        address indexed from,
+        address token,
+        uint256 tokenId
+    );
+
+    event ERC721Received(
+        bytes32 indexed msgHash,
+        address indexed from,
+        address indexed to,
+        uint256 srcChainId,
+        address token,
+        uint256 tokenId
+    );
+
+    error NFTVAULT_CANONICAL_TOKEN_NOT_FOUND();
+    error NFTVAULT_INVALID_OWNER();
+    error NFTVAULT_INVALID_SENDER();
+
+    function sendErc721(
```

Comment:
We have `sendErc721`, `sendErc1155`, `sendERC20`, I think we can simply call them the same name `sendToken`...  same for `releaseToken`, `receiveToken`.